prisma-fmt/src/lint.rs

```diff
@@ -12,19 +16,19 @@ pub struct MiniError {
 }
 
 pub(crate) fn run(schema: SchemaFileInput) -> String {
-    let schema = match schema {
+    let validated_schema = match schema {
         SchemaFileInput::Single(file) => psl::validate(file.into()),
         SchemaFileInput::Multiple(files) => psl::validate_multi_file(&files),
     };
-    let diagnostics = &schema.diagnostics;
+    let ValidatedSchema { diagnostics, db, .. } = &validated_schema;
 
     let mut mini_errors: Vec<MiniError> = diagnostics
         .errors()
         .iter()
         .map(|err: &DatamodelError| MiniError {
-            file_name: schema.db.file_name(err.span().file_id).to_owned(),
-            start: err.span().start,
-            end: err.span().end,
+            file_name: db.file_name(err.span().file_id).to_owned(),
+            start: offset_to_lsp_offset(err.span().start, db.source(err.span().file_id)),
+            end: offset_to_lsp_offset(err.span().end, db.source(err.span().file_id)),
```

Comment:
Should we have a function that takes a span and returns a pair of LSP offsets to avoid traversing the document in O(n) twice?

Reply:
I have implemented the `span_to_range` function. Additionally, I replaced a function with the same name and role in `offset.rs`(see https://github.com/prisma/prisma-engines/pull/4815/files#diff-557db087b8d611a0049284a4b38c817f1868a1dde965a9fa904c2dcaad576eb0R198). Since the order of arguments was different, I modified the usage accordingly. For consistency, I also adjusted the order of arguments for the `range_after_span` function. Since this is an internally used function, I believe there should be no issues.